payu/models/model.py

```diff
@@ -198,6 +177,59 @@ def get_prior_restart_files(self):
             print("No prior restart files found: {error}".format(error=str(e)))
             return []
 
+    def expand_executable_path(self, exec):
+        """Given an executable, return the expanded executable path"""
+        # Check if exe is already an absolute path
+        if os.path.isabs(exec):
+            return exec
+
+        # Check if path set by loading user modules has been defined
+        module_added_path = self.expt.user_modules_path
+        if module_added_path is None:
+            print("payu: warning: Skipping searching for model executable " +
+                  "in $PATH set by user modules")
+            module_added_paths = []
+        elif module_added_path == '':
+            module_added_paths = []
+        else:
```

Comment:
I'm not clear on the difference between these two cases, namely `None` or `''`. Don't they both indicate search will be skipped?

Reply:
In both cases we're going to fall through to the `return` condition right?

Reply:
As experiment now has a field `self.user_modules_paths` which is a set or None, I've updated to just check if it `None` with the error message. If loading user modules, and there were no changes to $PATH, `self.user_modules_paths` will just be an empty set()